library/std/src/sys/pal/unix/net.rs

```diff
@@ -213,16 +213,25 @@ impl Socket {
                 }
                 0 => {}
                 _ => {
-                    // linux returns POLLOUT|POLLERR|POLLHUP for refused connections (!), so look
-                    // for POLLHUP rather than read readiness
-                    if pollfd.revents & libc::POLLHUP != 0 {
-                        let e = self.take_error()?.unwrap_or_else(|| {
-                            io::const_io_error!(
-                                io::ErrorKind::Uncategorized,
-                                "no error set after POLLHUP",
-                            )
-                        });
-                        return Err(e);
+                    if cfg!(target_os = "vxworks") {
+                        // Check if the connnection actually succeeded and return ok only when
+                        // the socket is ready and no errors were found
+                        // https://github.com/rust-lang/rust/issues/127018
```

Comment:
Could you just add a note specifically saying vxworks does not set POLLHUP, POLLERR, or POLLOUT? Ok to delete the issue link with this if you want.

r=me after this

Reply:
r=me

Reply:
@tgross35 `r=me` doesn't work.

Reply:
Not literally `r=me`, It just means that someone can tell bors `r=tgross35` (or whoever said it) to approve it. But you still need to be in reviewers.

Reply:
Can I also be a Contributor/reviewer? What are the requirements? @tgross35

Reply:
Github adds the "contributor" label to anybody who has a merged PR, which you will soon, if that is what you meant. So welcome :). You can also always add your review to any PRs, help with issues, and join discussion on Zulip. You'll come across other ways to get more involved (WGs, teams) if you stay around for a while.

If you are interested in contributing more, something that would be immediately helpful is to list yourself as a maintainer for vxworks, since we could use somebody to refer to but don't have anyone listed yet. You just need to add a file to https://github.com/rust-lang/rust/tree/9c3bc805dd9cb84019c124b9a50fdff1e62a7ec9/src/doc/rustc/src/platform-support by duplicating `TEMPLATE.md` and updating it.